rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,344 @@
+In the same update it MUST register new proof and CA certificate as in the key loss process.
+Example update:
+```json
+{
+  "patch": [
+    { "op": "add", "path": "/prfs", "value": "some new proof" },
+    { "op": "replace", "path": "/crts", "value": ["CA certificate as base64 ASN.1"] }
+  ]
+}
+``` 
+
+If a vendor removes a certificate it SHALL NOT be re-added
+TODO: What if the attacker re-adds the certificate after the vendor removed it, so it can still sign 
+TODO: What if the attacker keeps adding anti-dated (timestamp in history) changes, branching from an earlier document before the certificate was revoked?
+TODO: Internal revocation list SHOULD be build when a certificate is removed
+
+### 4.2. Organization
+This document registers a care organization as a client of registered vendor (section 4.1).
+
+#### Fields
+The following fields MUST be present:
+* **id**: MUST contain the organization's ID encoded as URN. This can be any valid URN, for instance an AGB-code or
+  KVK-number e.g.: `urn:oid:2.16.840.1.113883.2.4.6.1:1234` (AGB-code). It is used to refer to the organization by the
+  vendor itself or other vendors. The organisation ID SHALL NOT be changed.
+* **vid**: (vendor ID) MUST be the organization's software vendor's ID as URN (e.g. `urn:oid:1.3.6.1.4.1.54851.4:1234`).
+  The vendor ID SHALL NOT be changed; relation to a care organization SHALL NOT be transfered to another vendor. 
+* **name**: MUST contain the organization's commonly known name as string. Only alphanumeric and space characters are allowed,
+  it SHALL NOT be empty or only contain spaces. It MUST conform to the following regex: `[a-zA-Z0-9 ]+`.
+* **prfs**: (proofs) MUST contain an array with cryptographic proofs asserting the organization as being an actual care organization.
+  at the same it serves as commitment from the care organization that it uses the vendor's (identified by **vid**) software.
+  See section 6 for supported proof types.
+
+Example:
+```json
+{
+    "id": "1df8a26c-77df-47d0-862c-c35cd18af86f",
+    "vid": "urn:oid:1.3.6.1.4.1.54851.4:1234",
+    "name": "HomeSweetHome Thuiszorg",
+    "prfs": ["organisation proof"]
+}
+```
+
+#### Document type
+The document type when indicating an organization MUST be `registry/organization`.
+
+#### Validation
+When processing an organization registration or update it MUST be validated as follows:
+
+1. Assert all required fields are present.
+2. Assert all fields value formats are valid.
+3. For updates: assert only modifiable fields are updated.
+4. Assert the document is signed by the vendor identified by **vid**.
+5. Validate **prfs**:
+   * Assert there is at least one proof.
+   * Assert proofs are valid at the time they're introduced.
+   * Assert all proofs authenticate the same organisation.
+
+### 4.3. Service
+This document registers a service provided by an organization and SHOULD describe the endpoints that SHOULD be used for a specified Bolt (use case).
+
+#### Fields
```

Comment:
service must be care org specific

Reply:
After discussion: endpoints are floating (made by a vendor), services created under a organization have references to endpoints.

TODO: Add domain model to this RFC